### ISSUE TITLE
Add more normalisation methods

### ISSUE THREAD

**Issue by nedtaylor** [nedtaylor/athena#4]
Include more methods of normalisation

- [ ] Change name of normalisation procedures
- [ ] Change 'norm' argument in renormalise_sum
- [ ] robust scaling (uses IQR instead of range from min-max scaling)
- [ ] log transformation

**Comment by nedtaylor** [comment]
Not sure anymore what is actually wanted here.
Most likely that we want to convert normalisation into preprocessing layers like Keras: https://keras.io/api/layers/preprocessing_layers/

- Resizing Layer
- Rescaling Layer
- Normalisation Layer
- Logarithmic Layer
- Discretisation Layer

**Comment by nedtaylor** [comment]
In the current form, we would need a layer for each dimensionality.

It'd be best if we just converted these ones into using rank 2 data (num_features, batch_size) and then, for get_output and get_gradient, we passed the rank that was assumed. It is meant to be a preprocessing layer, after all, so could just assume rank 2 and have it always feed into input1d_layer_type.

As other layers no longer take in assumed shapes, we should be able to handle this fine with mismatched ranks.
